bundles/net.tourbook/src/net/tourbook/tour/TourManager.java

```diff
@@ -2715,49 +2738,57 @@ private static void removeTourPauses(final TourData tourData,
-
-         return false;
+      if (tourDataList == null || tourDataList.size() == 0) {
+         return modifiedTours;
       }
 
-      final WeatherData historicalWeatherData = new HistoricalWeatherRetriever(tourData).retrieveHistoricalWeatherData().getHistoricalWeatherData();
-      if (historicalWeatherData == null) {
+      TourLogManager.showLogView();
+      final long start = System.currentTimeMillis();
 
-         TourLogManager.subLog_ERROR(NLS.bind(
-               Messages.Dialog_RetrieveWeather_WeatherDataNotFound,
-               TourManager.getTourDateTimeShort(tourData)));
+      final String weatherProvider = _prefStore.getString(
+            ITourbookPreferences.WEATHER_WEATHER_PROVIDER_ID);
 
-         return false;
-      }
+      TourLogManager.subLog_INFO(NLS.bind(
+            LOG_RETRIEVE_WEATHER_DATA_001_START,
+            weatherProvider));
 
-      tourData.setIsWeatherDataFromApi(true);
+      for (final TourData tourData : tourDataList) {
 
-      tourData.setAvgTemperature(historicalWeatherData.getTemperatureAverage());
-      tourData.setWeatherWindSpeed(historicalWeatherData.getWindSpeed());
-      tourData.setWeatherWindDir(historicalWeatherData.getWindDirection());
-      tourData.setWeather(historicalWeatherData.getWeatherDescription());
-      tourData.setWeatherClouds(historicalWeatherData.getWeatherType());
+         // ensure data is available
+         if (tourData.latitudeSerie == null || tourData.longitudeSerie == null) {
 
-      tourData.setWeather_Humidity((short) historicalWeatherData.getAverageHumidity());
-      tourData.setWeather_Precipitation(historicalWeatherData.getPrecipitation());
-      tourData.setWeather_Pressure((short) historicalWeatherData.getAveragePressure());
-      tourData.setWeather_Temperature_Max(historicalWeatherData.getTemperatureMax());
-      tourData.setWeather_Temperature_Min(historicalWeatherData.getTemperatureMin());
-      tourData.setWeather_Temperature_WindChill(historicalWeatherData.getWindChill());
+            TourLogManager.subLog_ERROR(String.format(
+                  LOG_RETRIEVE_WEATHER_DATA_010_NO_GPS_DATA_SERIE,
+                  getTourDateTimeShort(tourData)));
 
-      TourLogManager.subLog_OK(getTourDateTimeShort(tourData));
+            return modifiedTours;
+         }
 
-      return true;
+         BusyIndicator.showWhile(Display.getCurrent(),
+               () -> {
+
+                  if (TourWeatherRetriever.retrieveWeatherData(tourData, weatherProvider)) {
+
+                     modifiedTours.add(tourData);
+                     TourLogManager.subLog_OK(getTourDateTimeShort(tourData) +
+                           UI.SYMBOL_COLON + UI.SPACE +
+                           WeatherUtils.buildWeatherDataString(tourData, true, true));
+                  }
+               });
+      }
+
+      TourLogManager.subLog_INFO(String.format(
+            LOG_RETRIEVE_WEATHER_DATA_002_END,
+            (System.currentTimeMillis() - start) / 1000.0));
+
```

Comment:
Actually, with OpenWeatherMap, the result gives the hourly data so I will improve it by taking al the hours from the tour start to the tour end times.

Reply:
>Could it be possible to show also the weather station in the log?

OpenWeatherMap doesn't seem to give back the weather station in their API response. You can look at an example [here](https://github.com/mytourbook/mytourbook/blob/fb-improvements/bundles/net.tourbook.tests/src/test/data/weather/openweathermap/files/LongsPeak-Manual-OpenWeatherMapResponse.json) but I don't see such data.

Reply:
OpenWeatherMap weather retrieval is onw improved to only take the values included between the tour start and end times

Reply:
> Actually, with OpenWeatherMap, the result gives the hourly data so I will improve it by averaging all the data between the tour start to the tour end times.

Can you log all the hour/temperature values which are used to compute the average values, that someone know how it was calculated, otherwise this is a blackbox

Reply:
>>Can you log all the hour/temperature values which are used to compute the average values, that someone know how it was calculated, otherwise this is a blackbox

I can. Do you have a favorite format to log those values, or anything will do ?

Reply:
I added an option for the user to display the "full log" when retrieving weather data:

- The new preference option that is displayed for all the providers
![image](https://user-images.githubusercontent.com/25235665/157127058-4090929f-da2b-4cf0-8b62-bc5a83f9e809.png)

- An example of "full log" when using OpenWeatherMap
![image](https://user-images.githubusercontent.com/25235665/157127101-6f490902-1e7f-4dc9-b4c5-5f976ae57c4e.png)

Reply:
> > Could it be possible to show also the weather station in the log?
> 
> OpenWeatherMap doesn't seem to give back the weather station in their API response. You can look at an example [here](https://github.com/mytourbook/mytourbook/blob/fb-improvements/bundles/net.tourbook.tests/src/test/data/weather/openweathermap/files/LongsPeak-Manual-OpenWeatherMapResponse.json) but I don't see such data. However, it seems that [WWO gives the location information](https://github.com/mytourbook/mytourbook/blob/8bca5dee7aa5855afec3be2e959233d1329dab14/bundles/net.tourbook.tests/src/test/data/weather/worldweatheronline/files/LongsPeak-Manual-WorldWeatherOnlineResponse.json#L26) so I can add that for WorldWeatherOnline

I added the weather station information (for WorldWeatherOnline since OpenWeatherMap doesn't have such information) in the log when the "full log" is checked. The weather station information is a clickable link that opens the weather station website as the example below:

![image](https://user-images.githubusercontent.com/25235665/157166196-4531f13e-16d3-4c78-b8b1-81f646e8e05f.png)